pkg/client/jobs.go

```diff
@@ -0,0 +1,77 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	apiv1 "github.com/acorn-io/runtime/pkg/apis/api.acorn.io/v1"
+	"github.com/acorn-io/runtime/pkg/labels"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8slabels "k8s.io/apimachinery/pkg/labels"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (c *DefaultClient) JobGet(ctx context.Context, name string) (*apiv1.Job, error) {
+	job := &apiv1.Job{}
+	return job, c.Client.Get(ctx, kclient.ObjectKey{
+		Name:      name,
+		Namespace: c.Namespace,
+	}, job)
+}
+
+func (c *DefaultClient) JobList(ctx context.Context, opts *JobListOptions) ([]apiv1.Job, error) {
+	result := &apiv1.JobList{}
+
+	selector, err := k8slabels.Parse(fmt.Sprintf("%s=%s", labels.AcornManaged, "true"))
+	if err != nil {
+		return nil, err
+	}
+
+	if err = c.Client.List(ctx, result, &kclient.ListOptions{
+		Namespace:     c.Namespace,
+		LabelSelector: selector,
+	}); err != nil {
+		return nil, err
+	}
+
+	sort.Slice(result.Items, func(i, j int) bool {
+		if result.Items[i].CreationTimestamp.Time == result.Items[j].CreationTimestamp.Time {
+			return result.Items[i].Name < result.Items[j].Name
+		}
+		return result.Items[i].CreationTimestamp.After(result.Items[j].CreationTimestamp.Time)
+	})
+
+	if opts != nil && opts.App != "" {
+		var newResult []apiv1.Job
+		for _, job := range result.Items {
+			if job.Labels[labels.AcornAppName] == opts.App {
+				newResult = append(newResult, job)
+			}
+		}
+		return newResult, nil
+	}
+
+	return result.Items, nil
+}
+
+func (c *DefaultClient) JobDelete(ctx context.Context, name string) (*apiv1.Job, error) {
+	job, err := c.JobGet(ctx, name)
+	if apierrors.IsNotFound(err) {
+		return nil, nil
+	} else if err != nil {
+		return nil, err
+	}
```

Comment:
Right now we have to get the job so that we can return it at the end. When we do get that job and it doesn't exist it would be some extra work to then call a delete on it. Any thoughts on that?